hikari/internal/routes.py

```diff
@@ -190,16 +190,13 @@ class CDNRoute:
     path_template: str = attr.field()
     """Template string for this endpoint."""
 
-    valid_formats: typing.AbstractSet[str] = attr.field(
-        converter=_cdn_valid_formats_converter,
-        eq=False,
-        hash=False,
-        repr=False,
+    valid_formats: typing.Tuple[str, ...] = attr.field(
```

Comment:
Why switch to tuples when this is just used for checking whether x is in the valid formats (which is the main use case for sets)

Reply:
Check the pr description :)

Reply:
Why is memory size a concern for top level constants at that small scale

Reply:
Because, as stated above, they sum up to a lot of memory usage that can be avoided